pkg/tracing/event.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracing
+
+import (
+	"github.com/pingcap/dm/dm/pb"
+
+	"github.com/pingcap/errors"
+)
+
+// EventType represents trace event type
+type EventType byte
+
+// EventType list
+const (
+	EventNull EventType = iota
+	EventSyncerBinlog
+	EventSyncerJob
+	EventFlush // used to force upload tracing event
+)
+
+func (e EventType) String() string {
+	switch e {
+	case EventNull:
+		return "null"
+	case EventSyncerBinlog:
+		return "syncer binlog"
+	case EventSyncerJob:
+		return "syncer job"
+	case EventFlush:
+		return "flush"
+	default:
+		return "unknown event"
+	}
+}
+
+// Job is used for tracing evnets collecting and batch uploading
+type Job struct {
+	Tp    EventType
+	Event interface{}
+}
+
+// ProcessTraceEvents upload trace events to tracing service. ensure all jobs
+// have same EventType
+func (t *Tracer) ProcessTraceEvents(jobs []*Job) error {
+	defer t.rpcWg.Done()
```

Comment:
I think it's better to call `Done` **near** `Add`, can we change to that?

Reply:
fixed